docs/DeveloperGuide.md

```diff
@@ -626,7 +626,161 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+a whole range of fields which we offer. The fields could also be made optional depending on user's choice. 
+Their application will thus be initialized with user's very own fields which they require.
+
+**Examples**
+
+- User A could have the `Person` class with only `name`, `age`, `sex`, and `ic`.
+- User B could have the `Person` class with only `name`, `age`, `sex`,` ic` and `phone` with the `phone` field optional.
+
+
+### Flexible find command 
+
+**Potential Issues With Current Feature**
+
+Currently, the `find` command only allows users to find patient details using their `IC_NUMBER`. This could be inconvenient and troublesome 
+at times as users are required to type out the full `IC_NUMBER`. A single wrong character thus requires the user to re-look at the whole string of
+`IC_NUMBER` again in order to identify their mistakes.
+
+**Proposed Enhancements**
+
+In light of this issue, we would be working on our application to provide an update to the `find` command 
+to allow users to find contacts using different fields such as `name`, `age`, `email` etc. They will be case-insensitive to allow for better search results.
+
+**Examples**
+
+- find `John` returns contact with `name` of `John Doe` and corresponding `IC_NUMBER` of `T0123456A`
+- find `johndoe@mail.com` returns contact with `email` of `johndoe@mail.com` and corresponding `IC_NUMBER` of `T0123456A`
+
+### Update notes window during execution of `find` command
+
+**Potential Issues With Current Feature**
+
+Users have given us the feedback that our notes window on the right hand side of the split pane does not update with the use of `find` command.
+For instance, if our notes panel currently already have `John`'s notes reflected on it, using the `find` command to find someone who is not `John` will not cause the 
+notes panel to update accordingly. `John`'s notes will still exist on the notes panel despite the executing the `find` command to `find` a differing person.
+This could bring inconvenience to the user as users will not be able to view a contact's notes immediately upon using the `find` command. 
+They would then have to go through the hassle of typing the `show` command to allow the new contact's note to be displayed on the notes panel.
+
+**Proposed Enhancements**
+
+In order to make it more convenient for the users, we will enhance `find` command such that on executing the `find` command, the notes of the first 
+person appearing at the top of the search results will replace the current notes in the notes panel. We will also allow users to know whose note they are viewing, by displaying the patient's name or IC number.
+
+**Examples**
+
+- If the notes panel is currently displaying the notes of the user with `IC_NUMBER` of `T1234567C`, 
+executing `find T0123456A`, will cause the notes panel to update to the notes of user with `IC_NUMBER` of `T0123456A`
+
+### Enabling optional fields
```

Comment:
Maybe this one can merge with the first planned enhancement? because I feel like this is already mentioned a bit in the first enhancement.